Library/Homebrew/cmd/upgrade.rb

```diff
@@ -126,6 +126,9 @@ def upgrade_args
   def upgrade
     args = upgrade_args.parse
 
+    # Deprecated since this is now the default behavior.
+    # odeprecated "`brew upgrade --ignore-pinned`" if args.ignore_pinned?
```

Comment:
We should set the `--ignore-pinned` switch in the arg parses to `hidden` so it doesn't show up in man pages etc.